lib/hexpm/repository/releases.ex

```diff
@@ -166,25 +166,30 @@ defmodule Hexpm.Repository.Releases do
       "meta" => meta
     }
 
-    release = package && Repo.get_by(assoc(package, :releases), version: version)
-
-    multi
-    |> Multi.insert_or_update(:release, fn %{
+    case Version.parse(version) do 
+      :error ->
+        Ecto.Multi.error(multi, :version, "is not semantic")
```

Comment:
See comment below about tests and shape of errors.

If we return a changeset here then the rest of the system will know how to handle it. Not sure what's the nicest way to do that though, perhaps a schema less changeset like below?

```elixir
iex(1)> Ecto.Changeset.cast({%{}, %{version: Hexpm.Version}}, %{version: "1.0-dev"}, ~w(version)a)
#Ecto.Changeset<
  action: nil,
  changes: %{},
  errors: [version: {"is invalid", [type: Hexpm.Version, validation: :cast]}],
  data: %{},
  valid?: false
>
```

Thoughts?

Reply:
Yeah, I didn't feel good about wedging it in there like that and agree.